compiler/rustc_middle/src/mir/interpret/mod.rs

```diff
@@ -357,7 +350,7 @@ impl<'s> AllocDecodingSession<'s> {
                         bug!("this should be unreachable");
                     } else {
                         // Start decoding concurrently.
-                        sessions.insert(self.session_id);
+                        sessions.insert(0, self.session_id);
```

Comment:
This changes an O(1) operation to an O(n) operation. Should we be concerned?

(I don't know anything about this code, I am just looking at the diff)